src/menu/rendersettingsmenustate.h

```diff
@@ -1,11 +1,13 @@
 #pragma once
 
+#include "ui/widgets/tabbox.h"
```

Comment:
The issue with the compiler complaining about deleting an incomplete type came from an implicit destructor implementation. The `RenderSettingsMenuState` class doesn't explicitly declare a destructor, which means the destructor is synthetically generated as an inline destructor. By including the missing incomplete type, you solved that, but in turn made the `TabBox` an implicit, public dependency of this header, which is not necessary, as it's an implementation detail. The better approach would be to just declare a destructor in the header (`~RenderSettingsMenuState() override;`) and then force the implementation to be in the `.cpp` by instantiating the default implementation there (`RenderSettingsMenuState::~RenderSettingsMenuState() = default;`).